File: alexandria/ui/src/io/intino/alexandria/ui/displays/components/ImageAvatar.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.utils.AvatarUtil;

public class ImageAvatar<B extends Box> extends AbstractImageAvatar<B> {
	private String text;

	public ImageAvatar(B box) {
		super(box);
	}

	public String value() {
		return generateAvatar();
	}

	public ImageAvatar<B> text(String text) {
		this.text = text;
		return this;
	}

	public void update(String text) {
		this.text = text;
		refresh();
	}

	public void refresh() {
		notifier.refresh(value());
	}

	private String generateAvatar() {
		return AvatarUtil.generateAvatar(text, color());
	}

}

File: web-ui/src/displays/components/ImageAvatar.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractImageAvatar from "../../../gen/displays/components/AbstractImageAvatar";
import ImageAvatarNotifier from "../../../gen/displays/notifiers/ImageAvatarNotifier";
import ImageAvatarRequester from "../../../gen/displays/requesters/ImageAvatarRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';

const styles = theme => ({});

class ImageAvatar extends AbstractImageAvatar {

	constructor(props) {
		super(props);
		this.notifier = new ImageAvatarNotifier(this);
		this.requester = new ImageAvatarRequester(this);
	};


}

export default withStyles(styles, { withTheme: true })(ImageAvatar);
DisplayFactory.register("ImageAvatar", withStyles(styles, { withTheme: true })(ImageAvatar));